pkg/asset/quota/openstack/openstack.go

```diff
@@ -25,14 +41,37 @@ func Constraints(ci *validation.CloudInfo, controlPlanes []machineapi.Machine, c
 	var ret []quota.Constraint
 	ret = controlPlane(ci, ctrplConfigs)
 	ret = append(ret, compute(ci, computeReplicas, computeConfigs)...)
+	ret = append(ret, getNetworkConstraints(networkType)...)
 	return aggregate(ret)
 }
 
+func getNetworkConstraints(networkType string) []quota.Constraint {
+	if networkType == "kuryr" {
+		return []quota.Constraint{
+			{Name: "Port", Count: networkConstraintsWithKuryr.ports},
+			{Name: "Router", Count: networkConstraintsWithKuryr.routers},
+			{Name: "Subnet", Count: networkConstraintsWithKuryr.subnets},
+			{Name: "Network", Count: networkConstraintsWithKuryr.networks},
+			{Name: "SecurityGroup", Count: networkConstraintsWithKuryr.securityGroups},
+			{Name: "SecurityGroupRule", Count: networkConstraintsWithKuryr.securityGroupRules},
+		}
+	}
+	return []quota.Constraint{
+		{Name: "Port", Count: networkConstraints.ports},
+		{Name: "Router", Count: networkConstraints.routers},
+		{Name: "Subnet", Count: networkConstraints.subnets},
+		{Name: "Network", Count: networkConstraints.networks},
+		{Name: "SecurityGroup", Count: networkConstraints.securityGroups},
+		{Name: "SecurityGroupRule", Count: networkConstraints.securityGroupRules},
+	}
```

Comment:
nit: I think this makes it a bit easier to read and a bit more maintainable as you do not need to copy the translation logic.
```suggestion
	nc := networkConstraints
	if networkType == "kuryr" {
		nc = networkConstraintsWithKuryr
	}
	return []quota.Constraint{
		{Name: "Port", Count: nc.ports},
		{Name: "Router", Count: nc.routers},
		{Name: "Subnet", Count: nc.subnets},
		{Name: "Network", Count: nc.networks},
		{Name: "SecurityGroup", Count: nc.securityGroups},
		{Name: "SecurityGroupRule", Count: nc.securityGroupRules},
	}
```

Although, it is not immediately clear to me why we need the `networkConstraint` type at all. Why not just have the type of the `networkConstraints` and `networkConstraintsWithKuryr` variables be `quota.Constraint`. You could have a function that takes in as parameters the six fields of `networkConstraint` and returns a `quota.Constraint`.

```
var networkConstraint = buildNetworkConstraint(15, 1, 1, 1, 3, 60)
var networkConstraintWithKuryr = buildNetworkConstraint(1500, 1, 250, 250, 250, 1000)

func buildNetworkConstraint(ports, routers, subnets, networks, securityGroups, securityGroupRules int64) quota.Constraint {
	return []quota.Constraint{
		{Name: "Port", Count: ports},
		{Name: "Router", Count: routers},
		{Name: "Subnet", Count: subnets},
		{Name: "Network", Count: networks},
		{Name: "SecurityGroup", Count: securityGroups},
		{Name: "SecurityGroupRule", Count: securityGroupRules},
	}
}

func getNetworkConstraints(networkType string) []quota.Constraint {
	if networkType == "kuryr" {
		return []quota.Constraint{networkConstraintsWithKuryr}
	}
	return []quota.Constraint{networkConstraints}
}
```

Reply:
I'll apply your last suggestion, without the type. Thanks!